### DISCUSSION TITLE
createCardToken does not process special name values such as SECU, etc

### DISCUSSION THREAD

**Discussion by mschipperheyn** [mercadopago/sdk-js#71] [Q&A] [answered]
I moved from `cardForm` to manually processing cards, because `cardForm` is way too imperative. The `onSubmit` handler basically wants to dominate your form processing and cancels out any onSubmit handler you might have. Anyways, so I moved to `createCardToken`, expecting it would throw errors in the same way that the `onCardTokenReceived`, returns errors. To my surprise, instead of getting an error when passing a SECU, EXPI or other error related test name, it simply successfully returns a card token. How do I get similar processing results as with cardForm?

I would also REALLY appreciate it if this project would be managed in an open source fashion. Issues, source, etc. It's not fun to have a poorly documented project without source.

**Comment by mschipperheyn** [comment]
I get it now and got it working. The error returns and card token generation are separate. Makes sense.

**Reply by lucmantovani** [reply, marked as answer]
Hello @mschipperheyn!
I'm glad you was able to proceed with the implementation

With respect on how we manage this project, unfortunately we cannot open source it for security reasons
But you are always welcome on opening new discussions!

As for the issues, usually the most recommended channel is our developers' site support: [https://www.mercadopago.com/developers/support](https://www.mercadopago.com/developers/support)
